providers/dns/nifcloud/client.go

```diff
@@ -0,0 +1,202 @@
+// Package nifcloud implements a DNS provider for solving the DNS-01 challenge
+// using NIFCLOUD DNS.
+package nifcloud
+
+import (
+	"bytes"
+	"crypto/hmac"
+	"crypto/sha1"
+	"encoding/base64"
+	"encoding/xml"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+)
+
+const (
+	apiEndpoint = "https://dns.api.cloud.nifty.com"
+	apiVersion  = "2012-12-12N2013-12-16"
+	xmlns       = "https://route53.amazonaws.com/doc/2012-12-12/"
+)
+
+// ChangeResourceRecordSetsRequest is a complex type that contains change information for the resource record set.
+type ChangeResourceRecordSetsRequest struct {
+	Xmlns       string      `xml:"xmlns,attr"`
+	ChangeBatch ChangeBatch `xml:"ChangeBatch"`
+}
+
+// ChangeResourceRecordSetsResponse is a complex type containing the response for the request.
+type ChangeResourceRecordSetsResponse struct {
+	ChangeInfo ChangeInfo `xml:"ChangeInfo"`
+}
+
+// GetChangeResponse is a complex type that contains the ChangeInfo element.
+type GetChangeResponse struct {
+	ChangeInfo ChangeInfo `xml:"ChangeInfo"`
+}
+
+// ErrorResponse is the information for any errors.
+type ErrorResponse struct {
+	Error struct {
+		Type    string `xml:"Type"`
+		Message string `xml:"Message"`
+		Code    string `xml:"Code"`
+	} `xml:"Error"`
+	RequestID string `xml:"RequestId"`
+}
+
+// ChangeBatch is the information for a change request.
+type ChangeBatch struct {
+	Changes Changes `xml:"Changes"`
+	Comment string  `xml:"Comment"`
+}
+
+// Changes is array of Change.
+type Changes struct {
+	Change []Change `xml:"Change"`
+}
+
+// Change is the information for each resource record set that you want to change.
+type Change struct {
+	Action            string            `xml:"Action"`
+	ResourceRecordSet ResourceRecordSet `xml:"ResourceRecordSet"`
+}
+
+// ResourceRecordSet is the information about the resource record set to create or delete.
+type ResourceRecordSet struct {
+	Name            string          `xml:"Name"`
+	Type            string          `xml:"Type"`
+	TTL             int             `xml:"TTL"`
+	ResourceRecords ResourceRecords `xml:"ResourceRecords"`
+}
+
+// ResourceRecords is array of ResourceRecord.
+type ResourceRecords struct {
+	ResourceRecord []ResourceRecord `xml:"ResourceRecord"`
+}
+
+// ResourceRecord is the information specific to the resource record.
+type ResourceRecord struct {
+	Value string `xml:"Value"`
+}
+
+// ChangeInfo is A complex type that describes change information about changes made to your hosted zone.
+type ChangeInfo struct {
+	ID          string `xml:"Id"`
+	Status      string `xml:"Status"`
+	SubmittedAt string `xml:"SubmittedAt"`
+}
+
+// NifcloudDNSClient client of NIFCLOUD DNS
+type NifcloudDNSClient struct {
+	accessKey string
+	secretKey string
+}
+
+// NewDNSClient creates NifcloudDNSClient
+func NewDNSClient() *NifcloudDNSClient {
+	return &NifcloudDNSClient{
+		accessKey: os.Getenv("NIFCLOUD_ACCESS_KEY_ID"),
+		secretKey: os.Getenv("NIFCLOUD_SECRET_ACCESS_KEY"),
+	}
+}
```

Comment:
Could you remove `NewDNSClient` and move the client initialization to the  `NewDNSProvider`